### PR TITLE
Prevent node from starting across upgrades (until we support it better)

### PR DESCRIPTION
**Pull request by adagys**
On first run a version file is created in the node dir, and on subsequent runs the node version is matched against it.

### CODE REVIEW

**Comment by adagys** [comment]
Primary reviewer @Clintonio

**Comment by mikehearn** [comment]
Why is this a part of the caplet instead of the actual program? It'd be really useful to be able to feed the version number from gradle through into the build itself. If we ever move away from Capsule it'd suck to lose an important data integrity check.

**Comment by adagys** [comment]
My assumption here was that testnet/production deployments will be done via Capsule. If we're starting a node in any other way it means it's for demo/test/dev purposes and we don't really care about versioning or upgrades. But I guess it can't hurt to always do the check.

File: node/src/main/kotlin/net/corda/node/internal/Node.kt [hunk 1]
```diff
@@ -122,6 +124,27 @@ class Node(override val configuration: FullNodeConfiguration,
...
+    init {
+        checkVersionUnchanged()
+    }
+
+    /**
+     * Abort starting the node if an existing deployment with a different version is detected in the current directory.
+     * The current version is expected to be specified as a system property. If not provided, the check will be ignored.
+     */
+    private fun checkVersionUnchanged() {
+        val currentVersion = System.getProperty("corda.version") ?: return
```
Comment (adagys) [inline node/src/main/kotlin/net/corda/node/internal/Node.kt]: it's probably good to log a warning here before returning, but then we'd see it in all the demos (if run via Intellij) and tests

**Comment by adagys** [comment]
I don't really see why we'd do a custom solution if the packaging tool can provide it for free

**Comment by mikehearn** [comment]
It's in case we factor out some of the code that uses the node dir into a library some day, in which case the capsule wouldn't be present. But it's not an important concern right now. Can you delete branches from github once they're merged please? Thanks!

**Comment by adagys** [comment]
sorry, thought I did delete it!

**Pull request merged**